docs/auth/example-lh-auth.test.js

```diff
@@ -0,0 +1,134 @@
+    port: DEBUG_PORT,
+    onlyCategories: ['seo'],
+  });
+  return result.lhr;
+}
+
+describe('my site', () => {
+  /** @type {import('puppeteer').Browser} */
+  let browser;
+  /** @type {import('puppeteer').Page} */
+  let page;
+
+  beforeAll(async () => {
+    server.listen(SERVER_PORT);
+    browser = await puppeteer.launch({
+      args: [`--remote-debugging-port=${DEBUG_PORT}`],
+      headless: !process.env.DEBUG,
+      slowMo: process.env.DEBUG ? 50 : undefined,
+    });
+  });
+
+  afterAll(async () => {
+    await browser.close();
+    server.close();
+  });
+
+  beforeEach(async () => {
+    page = await browser.newPage();
+  });
+
+  afterEach(async () => {
+    await page.close();
+  });
+
+  describe('logged out', () => {
+    describe('homepage', () => {
+      beforeEach(async () => {
+        await page.goto('http://localhost:8000/');
+      });
+
+      it('lighthouse', async () => {
+        const lhr = await runLighthouse(page.url());
+        expect(lhr.categories.seo.score).toBeGreaterThanOrEqual(0.9);
+      });
+
+      it('login form should exist', async () => {
+        const emailInput = await page.$('input[type="email"]');
+        const passwordInput = await page.$('input[type="password"]');
+        expect(emailInput).toBeTruthy();
+        expect(passwordInput).toBeTruthy();
+      });
+    });
+
+    describe('dashboard', () => {
+      beforeEach(async () => {
+        await page.goto('http://localhost:8000/');
+      });
+
+      it('lighthouse', async () => {
+        const lhr = await runLighthouse(page.url());
+        expect(lhr.categories.seo.score).toBeGreaterThanOrEqual(0.9);
+      });
+
+      it('has no secrets', async () => {
+        expect(await page.content()).not.toContain('secrets');
+      });
+    });
+  });
+
+  describe('logged in', () => {
+    beforeEach(async () => {
+      const loginPage = await browser.newPage();
+      await loginPage.goto('http://localhost:8000/');
+      await loginPage.waitForSelector('input[type="email"]', {visible: true});
+
+      const emailInput = await loginPage.$('input[type="email"]');
+      await emailInput.type('admin@example.com');
+      const passwordInput = await loginPage.$('input[type="password"]');
+      await passwordInput.type('password');
+      const submitInput = await loginPage.$('input[type="submit"]');
+      await submitInput.press('Enter');
+      await loginPage.waitForNavigation();
+
+      await loginPage.close();
+    });
+
+    afterEach(async () => {
+      await page.goto('http://localhost:8000/logout');
+    });
+
+    describe('homepage', () => {
+      beforeEach(async () => {
+        await page.goto('http://localhost:8000/');
+      });
+
+      it('lighthouse', async () => {
+        const lhr = await runLighthouse(page.url());
+        expect(lhr.categories.seo.score).toBeGreaterThanOrEqual(0.9);
+      });
+    });
+
+    describe('dashboard', () => {
+      beforeEach(async () => {
```

Comment:
the layers of these `beforeEach` are a bit confusing to unravel for an example. we've got 3 different locations to read to figure out what state the page is in when these tests actually run. how would you feel about simplifying a bit so more of each test is directly laid out in an `it`?

Reply:
the inner describes aren't too useful, could removed those (they just navigate to the page in question).

removing the logged in/out describes would mean having a `login()` fn (L89-101). that sg?

Reply:
oh sorry missed this reply :)

yeah that sgtm!